src/plot_api/plot_api.js

```diff
@@ -385,6 +385,15 @@ function emitAfterPlot(gd) {
         fullLayout._redrawFromAutoMarginCount--;
     } else {
         gd.emit('plotly_afterplot');
+
+        // make the figure responsive
+        if(gd._context.responsive && !gd._responsiveChartHandler) {
+            // Keep a reference to the resize handler to purge it down the road
+            gd._responsiveChartHandler = function() {Plots.resize(gd);};
+
+            // Listen to window resize
+            window.addEventListener('resize', gd._responsiveChartHandler);
+        }
```

Comment:
Code looks great, but I'm a little surprised to see it so late in the sequence... `resize` has an internal delay, and calls `Plotly.relayout` which will wait for any pending drawing on the plot to finish before making its changes. So to avoid missing events that happen halfway through an async draw (mathjax or maps) I'd have thought we should put this in the synchronous block.

Reply:
It doesn't have to be this late. To be honest, I wasn't sure where it'd be best to put it in the sequence. Should I put at line 193?https://github.com/plotly/plotly.js/blob/8457b40375b7baae8a2722bfbc27263be6193dfb/src/plot_api/plot_api.js#L190-L196

Reply:
> Should I put it at line 193?

Seems like a good place!

Reply:
Oh one more thing occurs to me: a user could in principle start with `responsive` and then turn it off using `Plotly.react`... not sure why you'd want to do this, but it should work!